### ISSUE TITLE
Declaration files paths not being transformed when using 'import type...'

### ISSUE THREAD

**Issue by lukemorales** [LeDDGroup/typescript-transform-paths#73]
I've been struggling with the generated declaration files paths not being transformed to relative paths and I was using i.e ` import type { Type }  from '@/common/types'`.
I removed all `import type {}` and replaced with the standard `import { Type }  from '@/common/types'` and it transformed correctly in the declaration files, so there seems to be a problem resolving the `import type` syntax 🤔

**Comment by nonara** [comment]
Thanks for the report, Luke. What version of TS are you using?

**Comment by lukemorales** [comment]
@nonara TS version 4.0.3, latest available in VSCode

**Comment by nonara** [comment]
I can't reproduce on this end.

The following transforms properly for me:

`root-dir/file1.ts`  
```ts
export type A = string;
```

`index.ts`  
```ts
import type { A as ATypeOnly } from '#root/file1'
export { ATypeOnly }
```

Can you create a reproduction?

**Comment by nonara** [comment]
I've added tests to ensure type-only import works, and everything seems to be ok. Are you using v2+? 

If you can find a way to reproduce with two files, please let me know and I'll look into it.

**Comment by nonara** [comment]
Closing for housekeeping. Feel free to comment if you think this was in error.

**Comment by saqirmdevx** [comment]
Hello i got same issue here with latest version of Typescript and your plugin

<img width="877" alt="Screenshot 2022-08-20 at 14 09 22" src="https://user-images.githubusercontent.com/1737930/185745517-de8a360b-0560-43be-aa32-70881f8cd25e.png">

Javascript (Compiled import version)
<img width="764" alt="Screenshot 2022-08-20 at 14 11 12" src="https://user-images.githubusercontent.com/1737930/185745558-3a9df3b3-4faa-4f99-9590-4384759c04e8.png">

Basically "import type" remains in the file while it should be removed.

**Comment by patrickkunka** [comment]
Also seeing the above issue in the latest version

**Comment by danielpza** [comment]
@saqirmdevx Can you share your tsconfig.json configuration? What typescript and typescript-transform-paths versions are you using?

```
npm ls typescript
npm ls typescript-transform-paths
```

**Comment by saqirmdevx** [comment]
> @saqirmdevx Can you share your tsconfig.json configuration? What typescript and typescript-transform-paths versions are you using?
> 
> ```
> npm ls typescript
> npm ls typescript-transform-paths
> ```

I am sorry i already using different method (i created custom script that update paths after compilation)

I believe tsconfig is unchanged: (Paths might be different) 
```ts
{
  "compilerOptions": {
    /* Basic Options */
    "target": "ESNext" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */,
    "module": "ESNext" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */,
    "outDir": "build" /* Redirect output structure to the directory. */,
    "rootDir": "../" /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */,
    "removeComments": true /* Do not emit comments to output. */,
    /* Strict Type-Checking Options */
    "strict": true /* Enable all strict type-checking options. */,

    /* Additional Checks */
    "noUnusedLocals": true /* Report errors on unused locals. */,
    "noImplicitReturns": true /* Report error when not all code paths in function return a value. */,
    "noFallthroughCasesInSwitch": true /* Report errors for fallthrough cases in switch statement. */,

    /* Module Resolution Options */
    "moduleResolution": "node",
    "baseUrl": ".",
    "esModuleInterop": true,
  
    "paths": {
      "@/*": ["src/*"]
    }
  },
  "include": [
    "src/**/*.ts",
    "../shared/*.ts"
  ],
  "exclude": [ "../client/**"]
}
```

I was using latest typescript (stable version) and latest typescript-transform-paths.

**Comment by patrickkunka** [comment]
I am currently also experiencing the issue as described by @saqirmdevx whereby `import type ...` statements are present in JS output.

typescript: 4.8.4
typescript-transform-paths: 3.3.1

**Comment by nonara** [comment]
Thanks, all. Sounds like an issue with new TS. I'll have a look.

**Comment by patrickkunka** [comment]
Not sure if related, but I see the following warning in the console when I run `tsc` with the plugin applied:

```
DeprecationWarning: 'updateImportTypeNode' has been deprecated since v4.6.0. Use the overload that accepts 'assertions'
```

**Comment by nonara** [comment]
This issue has been resolved in v3.4!

**Comment by patrickkunka** [comment]
Brilliant, I will give it a whirl next week

**Comment by patrickkunka** [comment]
Apologies for the delay. I just tried version 3.4.4 but the issue still persists. There are `import type ...` statements in the compiled JS output.

**Comment by nonara** [comment]
@patrickkunka What version of TS? If it's 4.9, there's an issue affecting most transformers, including this. In that case it would not work at all.

If the problem happens with ts 4.8, if you can provide a quick repro, I will check it out. The issues reported in this thread were definitely corrected, so I'd have to dig in to see what you're facing.

**Comment by patrickkunka** [comment]
Hi @nonara it's still 4.8.4 as reported above
